pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -353,7 +353,6 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(
 			httpOpts: &httpListenerOpts{
 				rds:              routeName,
 				useRemoteAddress: true,
-				direction:        http_conn.HttpConnectionManager_Tracing_EGRESS, // viewed as from gateway to internal
```

Comment:
no replacement for this?

Reply:
replacement is TrafficDirection on listener -  which I think we are already setting.

Reply:
Can we validate this is true and we are setting this everywhere? otherwise we are going to break tracing and no one will notice for months

Reply:
> otherwise we are going to break tracing and no one will notice for months

It sounds like we need a test that validates/ensures that we have set the tracing config correctly and any conscious changes, need updating that test case?

Reply:
Do we have any test covering this?

Reply:
TrafficDirection definitely overrides this (I did this change).
I can't say that we set traffic direction everywhere (we should).

Reply:
Reg, Setting traffic direction on Listeners I verified some time back and ensured that it is set on all listeners - https://github.com/istio/istio/pull/18833 

 >  Do we have any test covering this?

Not sure. If it is not, I think it may be a good idea to add one that validates that we have the config needed for tracing functionality.

Reply:
Yeah